### PR TITLE
✨(front) implement front tests for ashley's editor

### PR DESCRIPTION
**Pull request by carofun**
## Purpose

We want to secure the render of our code by adding
additionnals front tests in the project. We use draft.js combined
with plugins, each one extends the feat of our editor. We
want to make sure each item is present as intended in the final render.

## Proposal

Adding tests that controls that the editor is rendered and expected elements are present such as :

- default toolbar plugin
- emoticon button
- link button
- quotation button

### CODE REVIEW

**Review by lunika — COMMENTED** [review]
You should split this PR in 2 commits. A first one for all the configuration and the second one dedicated to the test implementation.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
  - upgrade django-machina from 1.1.3 to 1.1.4
  - add button to insert quotes in the draftjs editor
+ - add frontend test for draft-js editor
```
Comment (lunika) [inline CHANGELOG.md:15]: It should not be in the `Changed` section but in the `Added` one.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ Versioning](https://semver.org/spec/v2.0.0.html).
```
Comment (lunika) [inline CHANGELOG.md:12]: Your commit title must be at the imperative, present tense. Can be `implement front tests for ashley's editor`.
The gitmoji used is not relevant IMO, you are deploying nothing. I think ✨ is a more appropriated.

Comment (carofun) [reply CHANGELOG.md:12]: ok noted, thanks for the reminder

**Review by jbpenrath — COMMENTED** [review]
I would also test that toolbar buttons render correctly entities.

File: src/frontend/js/components/AshleyEditor/index.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    });
+
+    // check if emoticon has been added to the toolbar plugin editor
+    screen.getByRole('button', {
+      name: /☺/i,
+    });
+
+    // check if quotation has been added to the toolbar plugin editor
+    const quotationButton = container.querySelector(
+      'path[d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"]',
```
[the remark this answers is not in the document]
Comment (carofun) [reply src/frontend/js/components/AshleyEditor/index.spec.tsx:26]: I've found nothing else ....

File: src/frontend/js/components/AshleyEditor/index.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    );
+    expect(quotationButton).toBeInTheDocument();
+
+    // check if link button has been added to the toolbar plugin editor
+    const linkButton = container.querySelector(
+      'path[d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"]',
```
[the remark this answers is not in the document]
Comment (carofun) [reply src/frontend/js/components/AshleyEditor/index.spec.tsx:32]: same here I've found nothing else than the svg of the searched button

**Comment by carofun** [comment]
> I would also test that toolbar buttons render correctly entities.

We've spoken about it but as these behaviors are rendered by plugins et those ones are tested, we've decided that this was already tested.
Can you confirm that I need to do so ?

File: src/frontend/webpack.config.js [hunk 5]
```diff
@@ -32,7 +32,9 @@ module.exports = {
         use: [
           {
             loader: 'babel-loader',
-            options: require('./babel.config'),
+            options: {
+              babelrc: true,
+            },
```
[the remark this answers is not in the document]
Comment (carofun) [reply src/frontend/webpack.config.js:37]: I'll give it a try

**Review by jbpenrath — APPROVED** [review]
LGTM ! 🎉

**Pull request merged**